src/components/EventCard.tsx

```diff
@@ -42,90 +54,157 @@ const EventCard = (event: Event): ReactElement => {
       })
     : [];
 
-  const [isRegistered, setIsRegistered] = useState(true);
+  //nuid of the corrently loggedIn user
+  const { userID } = useContext(LoginContext);
+  console.log(attendanceChange);
+  const [isRegistered, setIsRegistered] = useState(
+    attendanceChange ? false : true
+  );
+  const [errorType, setErrorType] = useState(0);
+  const [isOpen, setIsOpen] = useState(false);
+  const [createdAttendanceChange, setCreatedAttendanceChange] = useState({});
+
+  const openModal = () => {
+    // need to make sure they don't already have an existing attendanceChange for this event
+    isRegistered ? setIsOpen(true) : setIsOpen(false);
+    //it would be cool if there was an indicator that they cannot register again(idk)
+  };
+  const closeModal = () => {
+    setIsOpen(false);
+  };
 
   const regButtonStyle = isRegistered
     ? "button-base-white px-3 my-2 mr-5 w-32"
     : "button-base-red px-4 my-2 mr-5 w-32";
 
-  const toggleReg = () => {
-    setIsRegistered(!isRegistered);
-    // POST API Call Here
+  // for some great reason {} is not empty to adding this check before
+  const isObjectEmpty = (object: any) => {
+    for (const x in object) {
+      return false;
+    }
+    return true;
   };
 
+  //idk why but refreshing my dev server springs this useEffect, not sure if this is an issue though
```

Comment:
If you can remove these comments, that would be great. The only comments that should be here should be comments that are explanatory or TODOs, so if you have anything to fix, leave them as "// TODO"